pinot-core/src/main/java/org/apache/pinot/core/data/table/ConcurrentIndexedTable.java

```diff
@@ -125,50 +124,44 @@ public int size() {
   }
 
   private void resize(int trimToSize) {
-
     long startTime = System.currentTimeMillis();
-
-    _tableResizer.resizeRecordsMap(_lookupMap, trimToSize);
-
+    // when the resizer trims using a PQ, it will return a new trimmed map.
+    // the reference held by the indexed table needs to be updated. this is also
+    // the reason why it is volatile since the thread doing the resize will result in
+    // a new reference
+    _lookupMap = (ConcurrentMap)_tableResizer.resizeRecordsMap(_lookupMap, trimToSize);
```

Comment:
This is a perf enhancement. Reusing the map requires the TableResizer to invoke retainAll() on the existing map which was taking considerable period of time. PQL also builds a new map.

Reply:
PQL builds a new map because the new map is final and won't grow again. We cannot use `retainAll()` because looking up values in PQ is expensive, but we should try to use the existing map to avoid the overhead of growing the map.
I'm okay keeping it this way for now, but please add a TODO